### PR TITLE
Updated to 1.17

### PR DESCRIPTION
**Pull request by craftycorvid**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,3 +12,11 @@
 /run_server/*
 !/run_server/eula.txt
 /minecraft/
+
+# vscode
```
Comment (LoganDark) [inline .gitignore]: Please remove vscode stuff

**Review by LoganDark — CHANGES_REQUESTED** [review]
So there's a lot of noise in this PR, mostly gradle being updated, I can deal with that. Dunno why Java is updated all the way to 16, but if it works it works. Only got one nitpick here, good job.

**Comment by craftycorvid** [comment]
Thanks for the quick feedback. Minecraft 1.17 runs on Java 16 by default, hence the update. The gradle update is for the same reason, need to bump up to gradle 7 for Java 16 support.

File: build.gradle [hunk 2]
```diff
@@ -13,7 +13,7 @@ group = project.maven_group
 minecraft {}
 
 repositories {
-	maven { url = 'http://maven.fabricmc.net/' }
+	maven { url = 'https://maven.fabricmc.net/' }
```
Comment (LoganDark) [inline build.gradle:16]: Good catch here, no idea why it was insecure HTTP. :)

**Comment by LoganDark** [comment]
Have you tested this build? Surely 1.17 changed the translation system in some way. LanguageHack is a hack, after all.

**Comment by craftycorvid** [comment]
I ported languagehack to get this mod running on 1.17: https://github.com/sefodopo/FabricAutoBackup

I've been running it on my 1.17 server without issues.

**Comment by LoganDark** [comment]
> I ported languagehack to get this mod running on 1.17: https://github.com/sefodopo/FabricAutoBackup
> 
> I've been running it on my 1.17 server without issues.

Oh epic, I didn't know people actually depended on my mod to get their translations to load. Definitely merging this and making a release then. Thank you!

**Comment by LoganDark** [comment]
Built and published to maven: http://maven.logandark.net/net/logandark/languagehack/1.0.0-rc.4-1.17/

**Pull request merged**